src/presence_analyzer/tests.py

```diff
@@ -86,8 +156,89 @@ def test_get_data(self):
             datetime.time(9, 39, 5)
         )
 
+    def test_seconds_since_midnight(self):
+        """
+        Test time elapsed with different time set. Special case: midnight.
+        """
+        midnight_time = datetime.time(0, 0, 0)
+        second_after_midnight = datetime.time(0, 0, 1)
+        minute_after_midnight = datetime.time(0, 1, 0)
+        hour_after_midnight = datetime.time(1, 0, 0)
+        self.assertEqual(utils.seconds_since_midnight(midnight_time), 0)
+        self.assertEqual(utils.seconds_since_midnight(second_after_midnight), 1)
+        self.assertEqual(utils.seconds_since_midnight(minute_after_midnight), 60)
+        self.assertEqual(utils.seconds_since_midnight(hour_after_midnight), 3600)
+
+    def test_interval(self):
+        """
+        Test interval
+        """
+        start = datetime.time(0, 0, 0)
+        stop = datetime.time(1, 1, 1)
+        self.assertEqual(utils.interval(start, stop), 3661)
+
+    def test_mean_with_empty_list(self):
+        """
+        Mean checks whether provided list is empty and returns 0 if so.
+        """
+        self.assertEqual(utils.mean([]), 0)
+
+    def test_mean_with_non_empty_list(self):
+        """
+        Count mean and assert it provides 3-point accuracy.
+        """
+        items = [1, 2]
+        self.assertAlmostEqual(utils.mean(items), 1.5, 3)
 
-def suite():
+    def test_invalid_data_handled(self):
+        """
+        Captures log and checks whether 'Problem with line' message is found.
+        """
+        with testfixtures.LogCapture() as log_capture:
+
+            main.app.config.update({'DATA_CSV': INVALID_FORMAT_TEST_DATA})
+            try:
+                utils.get_data()
+            except UnboundLocalError:
+                pass
+            main.app.config.update({'DATA_CSV': TEST_DATA_CSV})
+            self.assertTrue(any(filter(lambda x:
+                                       'Problem with line' in x.msg, log_capture.records)))
```

Comment:
I don't like this + wrong styles. If u rly wanted to test it u should use mock like:
![image](https://cloud.githubusercontent.com/assets/5078402/24786736/1b82e94c-1b64-11e7-9da3-a900f71f4272.png)